### DISCUSSION TITLE
SQL Interpreter Question

### DISCUSSION THREAD

**Discussion by U-C-S** [SeaQL/summer-of-code#10] [General]
Greetings, I'm Chanakya, A 3rd year University CS Student and a Prospective contributor from India. I am bit excited and new to SeaORM codebase, though I know about this project since the release of v0.3. I'm currently working on to write proposal for `5. A SQL interpreter primarily intended for Mock testing` . I have previous ORM experience with EF Core and Prisma.

ADDITIONAL: 
I believe I have already have done something not-so-similar kind of [project](https://github.com/U-C-S/MiniDB) during my 2nd year, 9months ago (Ignore my past-self's poor use of words in README). 

A questions I have over the [idea](https://github.com/SeaQL/summer-of-code/tree/main/2022#5-a-sql-interpreter-primarily-intended-for-mock-testing)
1. I believe this will be replacing SQLite in https://www.sea-ql.org/SeaORM/docs/write-test/sqlite . Am I correct ??
2. if I'm correct, Could this be alternatively called a in-memory database ?? something like this from [efcore](https://docs.microsoft.com/en-us/ef/core/testing/testing-without-the-database#in-memory-provider)
3. If already discussed, can you link me to existing discussion on this idea, discord or github ?? 
4. Do you plan to use anywhere else of this testing ??

**Comment by tyt2y3** [comment]
1. No. It will be used to augment Mock testing
2. Yes, they are comparable. But we aim to solve all pitfalls of those mentioned in the EF link
3. Not quite
4. No, it will not be used in production

**Reply by U-C-S** [reply]
Sorry, I framed my `4.` question wrong. 

I meant to ask - Do you have any other areas where you plan to use it, outside of Mock testing ?

**Reply by tyt2y3** [reply]
Not in the scope of the project for now
